automation/src/test/scala/org/broadinstitute/dsde/test/api/SamApiSpec.scala

```diff
@@ -401,26 +401,115 @@ class SamApiSpec extends FreeSpec with BillingFixtures with Matchers with ScalaF
         5.minutes, 5.seconds)
 
       // Change the membership of the admin policy to include users 1 and 2
-      Sam.user.setPolicyMembers(managedGroupId, adminPolicyName, Set(user1.email, user2.email))(user1AuthToken)
+      Sam.user.setManagedGroupPolicyMembers(managedGroupId, adminPolicyName, Set(user1.email, user2.email))(user1AuthToken)
       awaitAssert(
         Await.result(googleDirectoryDAO.listGroupMembers(policyEmail.head), 5.minutes)
           .getOrElse(Set.empty) should contain theSameElementsAs Set(user1Proxy.value, user2Proxy.value),
         5.minutes, 5.seconds)
 
       // Add user 3 to the admin policy
-      Sam.user.addUserToPolicy(managedGroupId, adminPolicyName, user3.email)(user1AuthToken)
+      Sam.user.addUserToManagedGroupPolicy(managedGroupId, adminPolicyName, user3.email)(user1AuthToken)
       awaitAssert(
         Await.result(googleDirectoryDAO.listGroupMembers(policyEmail.head), 5.minutes)
           .getOrElse(Set.empty) should contain theSameElementsAs Set(user1Proxy.value, user2Proxy.value, user3Proxy.value),
         5.minutes, 5.seconds)
 
       // Remove user 2 from the admin policy
-      Sam.user.removeUserFromPolicy(managedGroupId, adminPolicyName, user2.email)(user1AuthToken)
+      Sam.user.removeUserFromManagedGroupPolicy(managedGroupId, adminPolicyName, user2.email)(user1AuthToken)
       awaitAssert(
         Await.result(googleDirectoryDAO.listGroupMembers(policyEmail.head), 5.minutes)
           .getOrElse(Set.empty) should contain theSameElementsAs Set(user1Proxy.value, user3Proxy.value),
         5.minutes, 5.seconds)
     }
+
+    "should only synchronize the intersection group for policies constrained by auth domains" in {
+      implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+      val authDomainId = UUID.randomUUID.toString
+      val Seq(inPolicyUser: Credentials, inAuthDomainUser: Credentials, inBothUser: Credentials) = UserPool.chooseStudents(3)
+      val inBothUserAuthToken = inBothUser.makeAuthToken()
+      val Seq(inAuthDomainUserProxy: WorkbenchEmail, inBothUserProxy: WorkbenchEmail) = Seq(inAuthDomainUser, inBothUser).map {
+        user => Sam.user.proxyGroup(user.email)(inBothUserAuthToken)
+      }
+
+      Sam.user.createGroup(authDomainId)(inBothUserAuthToken)
+      register cleanUp Sam.user.deleteGroup(authDomainId)(inBothUserAuthToken)
+
+      val authDomainPolicies = Sam.user.listResourcePolicies("managed-group", authDomainId)(inBothUserAuthToken)
+      val authDomainAdminEmail = for {
+        policy <- authDomainPolicies if policy.policy.memberEmails.nonEmpty
+      } yield {
+        policy.email
+      }
```

Comment:
@Qi77Qi curious to hear your opinion on for-comp vs `collect` here

Reply:
I'd prefer `collect`

Reply:
`collect` is my favorite

Reply:
but I thought _I_ was your favourite :'(

Reply:
https://github.com/typelevel/cats/blob/2f129c62ed5bd7d1f4d1b5847186716b25e6fb7f/core/src/main/scala/cats/TraverseFilter.scala  there's more :)